packages/solana-swap-find-provider/lib/SolanaSwapFindProvider.ts

```diff
@@ -0,0 +1,147 @@
+import { BigNumber, SwapParams, SwapProvider, Transaction } from '@liquality/types'
+import { Provider } from '@liquality/provider'
+import { PendingTxError } from '@liquality/errors'
+import { sha256 } from '@liquality/crypto'
+
+import _filter from 'lodash/filter'
+
+export default class SolanaSwapFindProvider extends Provider implements Partial<SwapProvider> {
+  private instructions = {
+    init: 0,
+    claim: 1,
+    refund: 2
+  }
+
+  async findInitiateSwapTransaction(swapParams: SwapParams): Promise<Transaction<any>> {
+    const { refundAddress } = swapParams
+
+    return await this._findTransactionByAddress({
+      address: refundAddress as string,
+      swapParams,
+      instruction: this.instructions.init,
+      validation: this._compareParams
+    })
+  }
+
+  async findClaimSwapTransaction(swapParams: SwapParams, initiationTxHash: string): Promise<Transaction<any>> {
+    const [initTransaction] = await this.getMethod('getParsedAndConfirmedTransactions')([initiationTxHash])
+
+    if (!initTransaction) {
+      throw new PendingTxError(`Transaction receipt is not available: ${initiationTxHash}`)
+    }
+
+    const {
+      _raw: { buyer }
+    } = initTransaction
+
+    return await this._findTransactionByAddress({
+      swapParams,
+      address: buyer,
+      instruction: this.instructions.claim,
+      validation: this._validateSecret
+    })
+  }
+
+  async findRefundSwapTransaction(swapParams: SwapParams, initiationTxHash: string): Promise<Transaction<any>> {
+    const [initTransaction] = await this.getMethod('getParsedAndConfirmedTransactions')([initiationTxHash])
+
+    if (!initTransaction) {
+      throw new PendingTxError(`Transaction receipt is not available: ${initiationTxHash}`)
+    }
+
+    const {
+      _raw: { seller }
+    } = initTransaction
+
+    return await this._findTransactionByAddress({
+      swapParams,
+      address: seller,
+      instruction: this.instructions.refund
+    })
+  }
+
+  async findFundSwapTransaction(): Promise<null> {
+    return null
+  }
+
+  _compareParams(
+    swapParams: SwapParams,
+    initTxParams: { buyer: string; seller: string; secret_hash: string; value: BigNumber; expiration: BigNumber }
+  ): boolean {
+    return (
+      swapParams.recipientAddress === initTxParams.buyer &&
+      swapParams.refundAddress === initTxParams.seller &&
+      swapParams.secretHash === initTxParams.secret_hash &&
+      swapParams.value.eq(initTxParams.value) &&
+      new BigNumber(swapParams.expiration).eq(initTxParams.expiration)
+    )
+  }
+
+  _validateSecret(swapParams: SwapParams, data: any): boolean {
```

Comment:
Use this one: https://github.com/liquality/chainabstractionlayer/blob/19697c1ca1ad2b3b7f304bb2b5d31690725e858a/packages/utils/lib/index.ts#L54